gui/dig.lua

```diff
@@ -0,0 +1,658 @@
+            end,
+            disabled = false,
+            show_tooltip = true,
+            on_activate = function()
+                self.dig_panel.prio = self.dig_panel.prio + 1
+                self.dig_panel:updateLayout()
+            end,
+        },
+        widgets.ToggleHotkeyLabel {
+            view_id = "autocommit_designation_label",
+            key = "CUSTOM_C",
+            label = "Auto-Commit: ",
+            active = true,
+            enabled = true,
+            disabled = false,
+            show_tooltip = true,
+            initial_option = true,
+            on_change = function(new, old)
+                self.dig_panel.autocommit = new
+                self.dig_panel.dirty = true
+            end,
+        },
+        widgets.HotkeyLabel {
+            view_id = "commit_label",
+            key = "CUSTOM_CTRL_C",
+            label = "Commit Designation",
+            active = true,
+            enabled = function()
+                return self.dig_panel.saved_cursor and self.dig_panel.mark
+            end,
+            disabled = false,
+            show_tooltip = true,
+            on_activate = function()
+                self.dig_panel:commit()
+            end,
+        } }
+end
+
+function GenericOptionsPanel:change_shape(new, old)
+    self.dig_panel.shape = shapes.all_shapes[new]
+    self.dig_panel:add_shape_options()
+    self.dig_panel:updateLayout()
+end
+
+--
+-- Dig
+--
+
+Dig = defclass(Dig, widgets.Window)
+Dig.ATTRS {
+    frame_title = "Dig",
+    frame = {
+        w = 47,
+        h = 40,
+        r = 2,
+        t = 18,
+    },
+    resizable = true,
+    resize_min = { h = 10 },
+    autoarrange_subviews = true,
+    autoarrange_gap = 1,
+    presets = DEFAULT_NIL,
+    shape = DEFAULT_NIL,
+    dirty = true,
+    prio = 4,
+    autocommit = true,
+    cur_shape = 1,
+}
+
+function Dig:init()
+    self:addviews { ActionPanel {
+        view_id = "action_panel",
+        get_mark_fn = function()
+            return self.mark
+        end,
+    },
+        GenericOptionsPanel {
+            view_id = "name_panel",
+            dig_panel = self,
+        } }
+end
+
+function Dig:postinit()
+    self.shape = shapes.all_shapes[self.subviews.shape_name:getOptionValue()]
+    if self.shape then
+        self:add_shape_options()
+    end
+end
+
+-- Add shape specific options dynamically based on the shape.options table
+-- Currently only supports 'bool' aka toggle and 'plusminus' which creates
+-- a pair of HotKeyLabel's to increment/decrement a value
+-- Will need to update as needed to add more option types
+function Dig:add_shape_options()
+    local prefix = "shape_option_"
+    if self.subviews ~= nil then
+        for i, view in ipairs(self.subviews) do
+            if string.sub(view.view_id, 1, string.len(prefix)) == prefix then
+                self.subviews[i] = nil
+            end
+        end
+    end
+    if self.shape then
+        if self.shape.options ~= nil then
```

Comment:
instead of successive nested `if`s, consider inverting the condition and returning from the function if true. e.g.
```lua
if not self.shape or not self.shape.options then return end
```

Reply:
Fixed